src/handler/mod.rs

```diff
@@ -552,6 +557,7 @@ impl Handler {
         let challenge_data = ChallengeData::try_from(packet.authenticated_data().as_slice())
             .expect("Must be the correct challenge size");
         debug!("Sending WHOAREYOU to {}", node_address);
+        self.add_expected_response(node_address.socket_addr);
```

Comment:
I can't right now tell if this is something that we were missing for an already existing `remove_expected_response` call, or if the call is missing after this change